File: android/src/main/java/com/reactnativerncustomerglu/RncustomergluPackage.java
package com.reactnativerncustomerglu;

import androidx.annotation.NonNull;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.reactnativerncustomerglu.Bannerwidget.Bannerwidget.BannerWidgetViewGroupManager;
import com.reactnativerncustomerglu.EmbedBannerwidget.EmbedBannerwidget.EBannerWidgetVGManager;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class RncustomergluPackage implements ReactPackage {
    @NonNull
    @Override
    public List<NativeModule> createNativeModules(@NonNull ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new RncustomergluModule(reactContext));
        return modules;
    }

    @NonNull
    @Override
    public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactContext) {
      return Arrays.asList(
        new EBannerWidgetVGManager(),
        new BannerWidgetViewGroupManager()
      );
    }
}

File: android/src/main/java/com/customerglu/reactnativecustomerglu/RnCustomergluPackage.kt
package com.customerglu.reactnativecustomerglu

import com.facebook.react.BaseReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider
import com.facebook.react.uimanager.ViewManager
import java.util.HashMap

class RnCustomergluPackage : BaseReactPackage() {
  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
    return if (name == RnCustomergluModule.NAME) {
      RnCustomergluModule(reactContext)
    } else {
      null
    }
  }

  override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
    return listOf(
      BannerViewManager(),
      CGEmbededViewManager()
    )
  }

  override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
    return ReactModuleInfoProvider {
      val moduleInfos: MutableMap<String, ReactModuleInfo> = HashMap()
      moduleInfos[RnCustomergluModule.NAME] = ReactModuleInfo(
        RnCustomergluModule.NAME,
        RnCustomergluModule.NAME,
        false,  // canOverrideExistingModule
        false,  // needsEagerInit
        false,  // isCxxModule
        true // isTurboModule
      )
      moduleInfos
    }
  }
}
